qa/task-cancellation/build.gradle

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.elasticsearch.gradle.Version
+import org.elasticsearch.gradle.VersionProperties
+import org.elasticsearch.gradle.info.BuildParams
+import org.elasticsearch.gradle.testclusters.RestTestRunnerTask
+
+apply plugin: 'elasticsearch.testclusters'
+apply plugin: 'elasticsearch.esplugin'
+apply from : "$rootDir/gradle/bwc-test.gradle"
+
+esplugin {
+  name 'task-cancellation'
+  description 'A test plugin that controls cross cluster requests so that we can verify the cancellation of them'
+  classname 'org.elasticsearch.plugin.tasks.TaskCancellationPlugin'
+}
+
+dependencies {
+  testImplementation project(':client:rest-high-level')
+}
+
+for (Version bwcVersion : BuildParams.bwcVersions.unreleasedWireCompatible) {
+  def currentVersion = VersionProperties.getElasticsearchVersion()
+  // TODO: Adjust the version after backporting
+  if (bwcVersion.before(Version.fromString("8.0.0"))) {
```

Comment:
I verified this against branch 7.x of my fork at https://github.com/dnhatn/elasticsearch/tree/7.x.